lib/solver.ml

```diff
@@ -98,20 +109,59 @@ let rec solve1 sol = function
   | Conj cs ->
      solve_conj sol cs
   | Disj cs ->
-     let issue_links = ["https://github.com/dwango/fialyzer/issues/99"] in
-     let message = "Solve disjunction of constraints" in
-     Error Known_error.(FialyzerError (NotImplemented {issue_links; message}))
+     solve_disj sol cs
 and solve_conj sol = function
   | [] -> Ok sol
   | c :: cs ->
      let open Result in
      solve1 sol c >>= fun sol' ->
      solve_conj sol' cs
+and solve_disj sol cs =
+  match List.filter_map ~f:(solve1 sol >>> Result.ok) cs with
+  | [] ->
+     let err0 = solve1 sol (List.hd_exn cs) in
+     err0
+  | sols ->
+     Ok (List.fold_left ~f:merge_solutions ~init:sol sols)
+
+(*
+  After calculating success typing, it is finally checked whether all branches pass type checking.
+  Note that this process is outside of the success typing algorithm.
+ *)
+let rec find_error_clauses sol = function
+  | Empty -> []
+  | Eq (ty1,ty2) ->
+     find_error_clauses sol (Subtype (ty1, ty2))
+     @ find_error_clauses sol (Subtype (ty2, ty1))
+  | Subtype (ty1, ty2) ->
+     begin match solve_sub sol ty1 ty2 with
+     | Ok _ -> []
+     | Error e -> [e]
+     end
+  | Disj cs ->
+     List.map ~f:(find_error_clauses sol) cs
+     |> List.concat
+  | Conj cs ->
+     find_conj sol cs
+and find_conj sol cs =
+  let f sol' c =
+    match (find_error_clauses sol' c, solve1 sol' c) with
+    | ([], Ok sol'') -> (sol'', [])
+    | ([], Error _)  -> failwith "cannot reach here"
+    | (es, Ok sol'') -> (sol'', es)
+    | (es, Error _)  -> (sol', es)
+  in
+  List.folding_map cs ~init:sol ~f
+  |> List.concat
 
 let rec solve sol cs =
   let open Result in
   solve1 sol cs >>= fun sol' ->
   if Map.equal (=) sol sol' then
-    Ok sol'
+    begin match find_error_clauses sol' cs with
+    | [] -> Ok sol'
+    | e :: _es ->
+       Error e
```

Comment:
I think it will be nice to see all errors.

Reply:
For the same reason of https://github.com/dwango/fialyzer/pull/127/files#r246673696, I simplified the display of errors.
I agree that all error message should be displayed. I will fix it.

Reply:
I see.

> I will fix it.

🙏

Reply:
Fixed.